project_stock/models/project_task.py

```diff
@@ -186,7 +186,11 @@ def write(self, vals):
         if "stage_id" in vals:
             stage = self.env["project.task.type"].browse(vals.get("stage_id"))
             if stage.done_stock_moves:
-                self.action_confirm()
+                # Avoid permissions error if the user does not have access to stock.
+                _self = self
+                if not self.env.user.has_group("stock.group_stock_user"):
+                    _self = self.sudo()
```

Comment:
Do sudo directly. This is not v12 where we have a trade-off of changing the user.